Yoast/javascript
# move social forms to separate package

## Summary


This PR can be summarized in the following changelog entry:

* Created a new package `social-metadata-forms`.
* Initialized yarn, babel and eslint.
* Moved `SocialMetadataPreviewForm`, `ImageSelect`, redux (reducer, actions, selectors), and tests to this new package.
 

## Relevant technical choices:

* Notify Plugin 2 that a new package is coming to prevent surprises during monorepo releases. (Do this when all is both packages are extracted.)


## Test instructions

This PR can be tested by following these steps:

* Check if the tests still work.
* Check the testApp if everything still works.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * Social Appearance
  * Snippet preview

## UI changes

## Quality assurance


Fixes #506
